library/src/main/java/com/klinker/android/send_message/Transaction.java

```diff
@@ -533,9 +533,15 @@ public static MessageInfo getBytes(Context context, boolean saveMessage, String[
         sendRequest.setDate(Calendar.getInstance().getTimeInMillis() / 1000L);
 
         try {
-            sendRequest.setFrom(new EncodedStringValue(Utils.getMyPhoneNumber(context)));
+            String phoneNumber = Utils.getMyPhoneNumberFromSubscription(context, settings.getSubscriptionId());
+            if (!TextUtils.isEmpty(phoneNumber)) {
+                sendRequest.setFrom(new EncodedStringValue(phoneNumber));
+            } else if (!TextUtils.isEmpty(fromAddress)) {
+                sendRequest.setFrom(new EncodedStringValue(fromAddress));
+            }
```

Comment:
You use this same block of code down below, as well. Could be a good idea to pull it into a function. `prepareFromAddress` or something

Reply:
I also think that the if/else blocks should be reversed. If the implementer is explicitly setting the `fromAddress`, then we should use that, instead of calculating it.

```java
if (!TextUtils.isEmpty(fromAddress)) {
    sendRequest.setFrom(new EncodedStringValue(fromAddress));
} else {
    String phoneNumber = Utils.getMyPhoneNumberFromSubscription(context, settings.getSubscriptionId());
    if (!TextUtils.isEmpty(phoneNumber)) {
        sendRequest.setFrom(new EncodedStringValue(phoneNumber));
    }
}
```

Reply:
Agree.

Reply:
> I also think that the if/else blocks should be reversed. If the implementer is explicitly setting the `fromAddress`, then we should use that, instead of calculating it.
> 
> ```java
> if (!TextUtils.isEmpty(fromAddress)) {
>     sendRequest.setFrom(new EncodedStringValue(fromAddress));
> } else {
>     String phoneNumber = Utils.getMyPhoneNumberFromSubscription(context, settings.getSubscriptionId());
>     if (!TextUtils.isEmpty(phoneNumber)) {
>         sendRequest.setFrom(new EncodedStringValue(phoneNumber));
>     }
> }
> ```

If we are able to read the number from system api, i believe we need to use this as priority. As there could be a potential false data set from the client (potential scenario could be say client asked user for entering the number and user might have entered a wrong number).